pubsub-light-plugin/src/main/java/org/jenkinsci/plugins/pubsub/InJenkinsEnvironmentContributor.java

```diff
@@ -0,0 +1,27 @@
+package org.jenkinsci.plugins.pubsub;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.EnvironmentContributor;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+/**
+ * Used to indicate to other components in this plugin that they are running in the context of a Jenkins instance.
+ */
+@Extension
+public class InJenkinsEnvironmentContributor extends EnvironmentContributor {
+    @Override
+    public void buildEnvironmentFor(@Nonnull final Run r, @Nonnull final EnvVars envs, @Nonnull final TaskListener listener) throws IOException, InterruptedException {
+        envs.put("IN_JENKINS", "true");
```

Comment:
@tfennelly added these to indicate to `PubsubBus.getBus()` that Jenkins is in use, to instantiate the correct impl of PubsubBus.

However, this approach seems brittle to me, since presumably it relies on Jenkins doing at least one job before this env gets set.  Also, based on what I've read, I'm not even sure this env variable will be visible in the same process.